<launch>

  <!-- GDB functionality -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

  <group ns="myrobot">

    <!-- Load hardware interface -->
    <node name="myrobot_hardware_interface" pkg="myrobot_ros_control" type="myrobot_node"
          output="screen" launch-prefix="$(arg launch_prefix)">
        <rosparam>
            period: 0.01     # 100 Hz
            write_skip: 100    # Write cycle of 100 Hz
        </rosparam>
    </node>

    <!-- Load controller settings -->
    <rosparam file="$(find myrobot_ros_control)/config/controllers.yaml" command="load"/>

    <!-- Load controller manager -->
    <node name="ros_control_controller_manager" pkg="controller_manager" type="controller_manager" respawn="false"
	  output="screen" args="spawn joint_state_controller position_trajectory_controller" />

    <!-- Convert joint states to /tf tranforms -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

  </group>

</launch>

